### ISSUE TITLE
Loud pop when first playing sound or after a longer pause

### ISSUE THREAD

**Issue by cshapeshifter** [wwmm/pulseeffects#179]
I'm running ```pulseeffects --gapplication-service &``` as part of ```.xinitrc```, I have pulseeffects enabled for all applications and only use the equalizer (to reduce some bass). It works well - thanks for this much better alternative to pulseaudio-equalizer.

The problem is that when I first play some sound, e.g. through mpd, on Youtube or wherever, there is a very, **very** loud pop coming from the speakers at the very start. The pop also happens when no sound has been playing for a while (not sure how long, a few minutes, maybe).

I don't have this problem when not using pulseeffects.

I don't know what useful info I could provide, but here's what the Wiki suggests:

Archlinux + XMonad
pulseeffects 3.2.1-1 installed via AUR (built from v3.2.1 GitHub archive)
Messages printed by PulseEffects when executed from command line: http://ix.io/TeH
The output of the command pactl list short: http://ix.io/TeJ
The output of the command pacmd list-sinks: http://ix.io/TeL
The output of the command pacmd list-sink-inputs: http://ix.io/TeN
The output of the command pacmd list-sources: http://ix.io/TeO
The output of the command pacmd list-source-outputs: http://ix.io/TeP
Screenshots of Pavucontrol tabs: I doubt these are relevant in this case.

**Comment by wwmm** [comment]
I see nothing wrong in your logs. Compared to mine the only difference is that you do not have a few plugins installed. But this should not be a problem. Not having them should be equivalent to disabling them and I hear no pop when I do that.

In order to save battery on laptop PE does not keep its GStreamer pipeline in the playing state all the time anymore. It checks if there is at least one app actually playing audio and alternates between the playing state and the ready state accordingly. it seems to me that the pop is happening whenever the GStreamer pipeline enters the playing state.

I am not sure about what we can do to identify the source of the problem but you can try:

- Running PE this way `GST_DEBUG=3 pulseeffects` and pay attention to what is printed when you have only one app shown in PE window and it starts playing. You can put higher numbers in `GST_DEBUG` but it really shows an overwhelming amount of logs if go too high.

- Execute `pacmd`. This will take to Pulseaudio prompt. There you can run `set-log-level 4`. This will put Pulseaudio in verbose mode and you will be able to track its messages with `journalctl -fb | grep -i pulseaudio`. Maybe Pulseaudio prints something suggestive whenever you pause / unpause an app like vlc. I don't think web browser will be good for this test unless you close the page and open it again. Pausing / unpausing a video in youtube does not have the same effect as doing the same in vlc.

- When PE is not running and you execute `gst-launch-1.0 audiotestsrc ! pulsesink` is there any pop?

Whatever we try our focus must be on what happens when GStreamer starts to play or recording audio from Pulseaudio. In PulseEffects it does both things at the same time!

**Comment by cshapeshifter** [comment]
Thanks for the help.

> When PE is not running and you execute gst-launch-1.0 audiotestsrc ! pulsesink is there any pop?

Yes there is!

I don't think this problem is related to pulseeffects anymore. I mistakenly thought so because I updated my system at the same time. Must have been an update, maybe to pulseaudio. There's even a POP when I just start pavucontrol after a longer pause. I'll close the issue but post an update when I solved the problem for future reference.
